stable_baselines/gail/dataset/record_expert.py

```diff
@@ -0,0 +1,138 @@
+import os
+import warnings
+
+import cv2
+import numpy as np
+from gym import spaces
+
+from stable_baselines.common.vec_env import VecEnv, VecFrameStack
+
+
+def generate_expert_traj(model, save_path, n_timesteps=0,
```

Comment:
Small suggestion, why not do this:
```python
def generate_expert_traj(model, save_path, env=None, n_timesteps=0,
                         n_episodes=100, image_folder='recorded_images'):
    """
    Train expert controller (if needed) and record expert trajectories.
    .. note::
        only Box and Discrete spaces are supported for now.
    :param model: (RL model or callable) The expert model, if it needs to be trained,
        then you need to pass ``n_timesteps > 0``.
    :param env: (gym env) The environment, if not defined then tries to use the model 
        environment.
    :param save_path: (str) Path without the extension where the
        expert dataset will be saved (ex: 'expert_cartpole' -> creates 'expert_cartpole.npz')
    :param n_timesteps: (int) Number of training timesteps
    :param n_episodes: (int) Number of trajectories (episodes) to record
    :param image_folder: (str) When using images, folder that will be used to record images.
    """
    if env is None and isinstance(model, BaseRLModel):
        env = model.get_env()

    assert env is not None, "You must set the env in the model or define it as a hyperparameter."

    ...    

    if n_timesteps > 0 and isinstance(model, BaseRLModel):
        model.learn(n_timesteps)

    ...

        if isinstance(model, BaseRLModel):
            action, _ = model.predict(obs)
        else:
            action = model(obs)
        obs, reward, done, _ = env.step(action)
```

it will allow both RL model use, and callable functions that can be PID models, real players, etc...

With an update to the doc with an example as well if you agree.

Reply:
Done in 78bfa30